### ISSUE TITLE
[Bug]: Double entries

### ISSUE THREAD

**Issue by jkemmerer1967** [mampfes/hacs_waste_collection_schedule#3121]
### I Have A Problem With:

The integration in general

### What's Your Problem

When including the ICS File -> HA Calendar is showing each entry twice.
When I open the ICS File in my e.g. Phone Calendar everything looks correct.
![double entries](https://github.com/user-attachments/assets/daf12d63-158e-4726-8006-194d9f218e25)

### Source (if relevant)

https://abfallkalender.regioit.de/kalender-moerfelden-walldorf/downloadfile.jsp?format=ics&zeit=-%3A00%3A00&lang=&jahr=2024&ort=Mörfelden-Walldorf&strasse=8475&fraktion=0&fraktion=1&fraktion=2&fraktion=5&fraktion=9&fraktion=10&fraktion=11&fraktion=12&md=-

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by respecreload** [comment]
<img width="589" alt="Bildschirmfoto 2024-12-02 um 09 35 05" src="https://github.com/user-attachments/assets/9d1de091-6c55-4b1e-9faa-67362c2127e0">

not really sure, when this happened, but I have the same issue here with a different source.

**Comment by 5ila5** [comment]
That is weird, I cannot reproduce this. Is there any specific configuration you changed? Is there something in your logs (settings -> system -> logs)? Does a HA restart resolve this (maybe not clearing last fetch correctly)?

**Comment by respecreload** [comment]
I tried to look into this issue in the past days and at least for my source, it seems like a problem with the python script that checks the data coming from the source. maybe with a newer python version something changed? anyways, I have seen an updated has been done to my source so this might be fixed in the future.

**Comment by JayAy1970** [comment]
Just installed the integration and have the double listing too.  Not sure how to specify the data source for you other than UK, Dacorum.

**Comment by JayAy1970** [comment]
Here 'are the Attributes returned in the different Details Formats which may point to where the problems are?

**Upcoming**
Upcoming data:
2025-03-24
Grey bin and kerbside caddy, Grey bin and kerbside caddy, Green bin, Green bin
2025-03-31
Blue bin and kerbside caddy, Blue bin and kerbside caddy

**Appointment**
Blue bin and kerbside caddy
March 31, 2025
Green bin
March 24, 2025
Grey bin and kerbside caddy
March 24, 2025

**Generic**
Types
Blue bin and kerbside caddy, Green bin, Grey bin and kerbside caddy
Upcoming
- date: '2025-03-24'
icon: mdi:trash-can
picture: null
type: Grey bin and kerbside caddy
- date: '2025-03-24'
icon: mdi:trash-can
picture: null
type: Grey bin and kerbside caddy
- date: '2025-03-24'
icon: mdi:leaf
picture: null
type: Green bin
- date: '2025-03-24'
icon: mdi:leaf
picture: null
type: Green bin
- date: '2025-03-31'
icon: mdi:recycle
picture: null
type: Blue bin and kerbside caddy
- date: '2025-03-31'
icon: mdi:recycle
picture: null
type: Blue bin and kerbside caddy
Last update
March 22, 2025 at 11:52:22

Upcoming data:
2025-03-24
Grey bin and kerbside caddy, Grey bin and kerbside caddy, Green bin, Green bin
2025-03-31
Blue bin and kerbside caddy, Blue bin and kerbside caddy

**Comment by 5ila5** [comment]
I really cannot reproduce this issue. I used <https://abfallkalender.regioit.de/kalender-moerfelden-walldorf/downloadfile.jsp?format=ics&zeit=-:00:00&lang=de&jahr=2025&ort=Mörfelden-Walldorf&strasse=8172&fraktion=0&fraktion=1&fraktion=2&fraktion=3&fraktion=4&fraktion=5&fraktion=6&fraktion=7&fraktion=8&fraktion=9&fraktion=10&fraktion=11&fraktion=12&md=-> and don't see any duplicates, can someone please provide an ICS file where this happens, I think the duplicate entries need to be inside the ICS file

**Comment by ChrSchwarz** [comment]
Hello,
Has anyone found a solution to this problem? I have the same problem, where my waste collection calendar entries appear twice. When I import the ICS calendar file, the appointments only appear once, and when I view the ICS file with an editor, all entries are also only present once. However, when I enter the URL in wastecollection, I get the duplicate entries.

https://abfallapp.softwareentwicklung-roth.de/{%Y}/KL/Mackenbach/Schubertstrasse/ics/de?abfallart_Restmuell=on&abfallart_GelberSack=on&abfallart_Biomuell=on&abfallart_Papier=on

**Comment by nebel969** [comment]
I have the same problem with the calendar from the Kaiserslautern district (e.g. [https://abfallapp.softwareentwicklung-roth.de/2026/KL/Hochspeyer/Enkenbacher_Strasse/ics/de?abfallart_Restmuell=on&abfallart_GelberSack=on&abfallart_Biomuell=on&abfallart_Papier=on](https://abfallapp.softwareentwicklung-roth.de/2026/KL/Hochspeyer/Enkenbacher_Strasse/ics/de?abfallart_Restmuell=on&abfallart_GelberSack=on&abfallart_Biomuell=on&abfallart_Papier=on)). I have already tried various towns and streets, but I always get duplicate entries. When importing into iCloud, each entry appears only once, as it should.

**Comment by int2018** [comment]
Not exactly but somehow I have a similar problem.
During setup I chose only specific Müll from the awsh.de source but when I check the calendar I see also other entries:

<img width="1026" height="486" alt="Image" src="https://github.com/user-attachments/assets/fad2138a-2358-4e89-a5f9-6dbe03d053f1" />

As a workaround the trigger uses the exact names but I wonder why the entries are imported although I did not check them.

**Comment by markvp** [comment]
We've investigated this again and cannot reproduce the duplicates. Testing with the Kaiserslautern ICS URL provided by @ChrSchwarz:

- The ICS file contains 66 unique events (no duplicates)
- The ICS parser returns exactly 66 entries with no duplicates

The most common cause of doubled entries is having the **same source configured twice** — for example:
- Both a YAML config and a UI config for the same source
- The source listed twice in `configuration.yaml`
- Two integration entries in Settings → Devices & Services

**To those still experiencing this:** Please check:
1. Do you have *both* a YAML config (`configuration.yaml`) and a UI-configured instance? If so, remove one.
2. In Settings → Devices & Services → Waste Collection Schedule, do you see multiple entries for the same source?
3. Can you share your full `configuration.yaml` waste_collection_schedule section (with addresses redacted)?

If someone can provide a reproducible case (ICS file + config that produces duplicates), we're happy to investigate further. Otherwise we'll close this as a configuration issue.

@int2018 Your issue (seeing unselected waste types) is different from duplicates — please open a separate issue with your full configuration so we can look into it.

**Comment by ChrSchwarz** [comment]
@markvp 
As we moved from 2025 to 2026, it suddenly started working as expected—no more duplicate entries. As far as I'm concerned, this can be closed. Thank you very much.

**Comment by markvp** [comment]
Thanks for confirming @ChrSchwarz — glad it resolved itself. The year rollover likely cleared a stale cache or duplicate calendar subscription.

@int2018 — the calendar view always shows all collection types from the source. Type filtering only applies to the **sensors**, not the calendar. This is by design. If you want to limit what appears, use the sensor `types` configuration to filter specific waste types.

Closing as resolved.

**Comment by int2018** [comment]
> [@int2018](https://github.com/int2018) — the calendar view always shows all collection types from the source. Type filtering only applies to the **sensors**, not the calendar. This is by design. If you want to limit what appears, use the sensor `types` configuration to filter specific waste types.
> Closing as resolved.

Thanks for the hint, that explains it.
